nav2_behavior_tree/plugins/condition/distance_traveled_condition.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2019 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__DISTANCE_TRAVELED_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__DISTANCE_TRAVELED_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "behaviortree_cpp_v3/condition_node.h"
+#include "nav2_util/robot_utils.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "tf2_ros/buffer.h"
+
+namespace nav2_behavior_tree
+{
+
+class DistanceTraveledCondition : public BT::ConditionNode
+{
+public:
+  DistanceTraveledCondition(
+    const std::string & condition_name,
+    const BT::NodeConfiguration & conf)
+  : BT::ConditionNode(condition_name, conf),
+    distance_(1.0),
+    first_time_(true)
+  {
+    getInput("distance", distance_);
+    node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+    tf_ = config().blackboard->get<std::shared_ptr<tf2_ros::Buffer>>("tf_buffer");
+  }
+
+  DistanceTraveledCondition() = delete;
+
+  BT::NodeStatus tick() override
+  {
```

Comment:
If you set to true first_time_ when the node is ticked and idle, then the node will get initialized again after being halted.

Reply:
Yes, going to follow pretty much the thing you did in the GoalUpdated condition